plugins/inventory/nb_inventory.py

```diff
@@ -1369,6 +1369,13 @@ def _fill_host_variables(self, host, hostname):
         if extracted_primary_ip:
             self.inventory.set_variable(hostname, "ansible_host", extracted_primary_ip)
 
+        if self.dns_name:
```

Comment:
I would structure this the same as the ones above and below it.

```suggestion
        if self.ansible_host_dns_name:
            extracted_dns_name = self.extract_dns_name(host=host)
            if extracted_dns_name:
                self.inventory.set_variable(
                    hostname, "ansible_host", extracted_dns_name
                )
```

Can you also update the tests as well? You can just update one of the test files for this new option to make sure it's setting the `ansible_host` with what is fetched as the `dns_name`.